### DISCUSSION TITLE
Synthmorph - Label maps generation

### DISCUSSION THREAD

**Discussion by EvanBeal** [voxelmorph/voxelmorph#360] [Q&A] [answered]
Hello !

Thank you for making available this nice library to perform image registration. It looks very promising and the approach used in Synthmorph seems really interesting to allow images of all contrasts/modalities to be registered together.

I would like to get started with Synthmorph and try to train a model following your methodology, however I can't find what  part of the code allows to generate input label maps from noise distribution. It seems that this is not included in the `train_synthmorph.py` function, therefore I was wondering how this should be done and if anyone could help me find what part of the code is involved in this process ?

Thanks in advance for the time you will take to answer me.

**Comment by mu40** [comment, marked as answer]
Hi @EvanBeal, And thanks for your interest in SynthMorph. Please have a look at our new demo at https://synthmorph.voxelmorph.net, which will take you through the steps of generating random label maps and using them to train a network. I realize we should get this added to the README promptly!

**Reply by EvanBeal** [reply]
Oh thank you really much. 
I wasn't aware of it and it seems indeed to be very helpful and well explained ! It will be really interesting to go through it.
